### PR TITLE
[jit] Add Span<T> intrinsics.

### PR DESCRIPTION
**Pull request by vargaz**
Part of #7249

### CODE REVIEW

File: mono/mini/intrinsics.c [hunk 1]
```diff
@@ -166,6 +179,65 @@ llvm_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
...
+
+		MonoGenericClass *gclass = mono_class_get_generic_class (cmethod->klass);
+		MonoClass *param_class = mono_class_from_mono_type (gclass->context.class_inst->type_argv [0]);
+
+		int span_reg = args [0]->dreg;
+		/* Load _pointer.Value */
+		int base_reg = alloc_preg (cfg);
+		EMIT_NEW_LOAD_MEMBASE (cfg, ins, OP_LOAD_MEMBASE, base_reg, span_reg, ptr_field->offset - sizeof (MonoObject));
+		/* Similar to mini_emit_ldelema_1_ins () */
+		if (mini_is_gsharedvt_variable_klass (param_class))
```
Comment (jaykrell) [inline mono/mini/intrinsics.c]: Emit one instruction and then fail?

Comment (vargaz) [reply mono/mini/intrinsics.c]: Fixed.

**Comment by kumpera** [comment]
Hey Zoltan, we should look into why ABCREM doesn’t work.

**Comment by vargaz** [comment]
Because the different references to the length of the span are based on different ldaddr opcodes.

**Comment by vargaz** [comment]
@monojenkins build Linux i386

File: mono/mini/intrinsics.c [hunk 2]
```diff
@@ -166,6 +179,74 @@ llvm_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
...
+emit_span_intrinsics (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsig, MonoInst **args)
+{
+	MonoInst *ins;
+
+	MonoClassField *ptr_field = mono_class_get_field_from_name (cmethod->klass, "_pointer");
+	if (!ptr_field)
+		/* Portable Span<T> */
+		return NULL;
+
+	if (!strcmp (cmethod->name, "get_Item")) {
```
Comment (kumpera) [inline mono/mini/intrinsics.c]: Don't we need to intrinsify set_Item as well?

Comment (vargaz) [reply mono/mini/intrinsics.c]: There is no set_Item, it returns a ref.

Comment (kumpera) [reply mono/mini/intrinsics.c]: So with Span ```
Span<int> span = ...;
span[0] = 10;
```

The store is compiled to this: ```
ldloc span
ldc.i4 10
call Span<int>::get_Item (int32)
stind.i4
```

**Comment by marek-safar** [comment]
@monojenkins backport 2018-06

**Pull request merged**